Refactor the code base & clean-up of main

Teufelchen1 opened the pull request:
Hi, 

this PR introduces major changes to the structure of the repository. 
It moves various code into theses subdirectories:
- `Sources/ftp`
- `Sources/menu`
- `Sources/net`
- `Sources/render`
- `Sources/sys`

It also changes the way NevolutionX builds by introducing new makefiles and including them. Note, one could have used other make strategies i.e. recursive make calls. I find my approach of simple concatenation of the `SRC` variable to be the easiest and straightforward choice but I'm open for alternatives. 

The code compiles and should run(didn't tested it), however, I don't think this is the best NevoX can be, so it will be a draft for further enhancements.

Teufelchen1 commented on .gitignore:
```diff
@@ -1,8 +1,10 @@
 DejaVuSansMono.ttf
 build/
 bin/
+*.DS_Store
```
mhm, shouldn't be here, added by accident, but it *could* stay as it does not hurt. You should decide about it.

dracc replied:
Let it stay. :) I do however feel like you should remove the .ttf from this list if you're touching it anyways.

dracc commented on Sources/render/Makefile:
```diff
@@ -0,0 +1 @@
+SRCS += $(SRCDIR)/render/font.cpp $(SRCDIR)/render/outputline.cpp $(SRCDIR)/render/renderer.cpp
```
Small typo, should be `camelCase` on outputLine.cpp. :)

dracc replied:
File should end with newline. :)

dracc commented on Sources/render/Makefile:
```diff
@@ -0,0 +1 @@
+SRCS += $(SRCDIR)/render/font.cpp $(SRCDIR)/render/outputline.cpp $(SRCDIR)/render/renderer.cpp
```
I think I'd prefer if `outputLine` lived in `debug` or similar instead of `render`.

Teufelchen1 replied:
Thought about that as well, agreed. Would also opt for changing name of function+file as one could think "ah this is used for regularly displaying stuff" instead of "this is for debug purpose only". One should also evaluate if it could be completely optional.

Teufelchen1 commented:
![nevox](https://user-images.githubusercontent.com/9516484/82373160-71ab4400-9a1d-11ea-8e22-5ea009d837fb.png)
adding this to the draft, to keep all discussion together. 
It shows a visualisation of includes between the 'submodules'.

dracc commented on .gitignore:
```diff
@@ -1,8 +1,10 @@
 DejaVuSansMono.ttf
```
Might as well remove this old font file too, especially as it's not part of the project anymore. :)

Teufelchen1 commented:
The latest commit builds but I would not recommend running it on real hardware(if it runs at all). I changed major setup routines, introduced a new thread, touched the video setup - all without giving it a second thought. It's wip.

Teufelchen1 commented:
Latest commit heavily cleaned up the main.cpp by moving a lot of app setup into sub functions.
It also moved the menu-logic into its own class. Code compiles fine but is not tested. I would heavily recommend to test in xqemu prior testing on hardware(but I have doubts the code even works as intended). 
This is again not final, its a suggestion and it also try's to lay down some basement work to integrates trimbors work later on. I will consider this Draft-Blocked until I got some feed back from you & we discussed how to continue. 💪 Can't wait to see NevoXs future!

dracc commented on Sources/ftp/ftpConnection.cpp:
```diff
@@ -3,7 +3,7 @@
 #include <algorithm>
 #include <map>
-#include "outputLine.h"
+#include "../debug/outputLine.h"
```
I'd very much prefer if we used paths relative to `Includes` (where I feel like "public" headers should live) over these paths starting with `..`.

Teufelchen1 commented:
Just realised I never tested the ftp 😨

Teufelchen1 commented:
@dracc Are you done fetching ideas out of this? It was good while it lasted but I think its time to close this one now.

dracc commented:
I'll make sure to revisit this for inspiration later. Thank you for your hard work!
